NSS-Therapeutic-Raccoons/BangazonSite
# Tg 2 List Products for Product Type

## Link to Ticket
Closes # [2](https://github.com/NSS-Therapeutic-Raccoons/BangazonSite/issues/2)

## Description of Proposed Changes
- Given the user is on any page of the Bangazon site
- When the user performs a gesture on a product category name
- Then a list of all products in that category will be displayed, with title, quantity, and price/unit

## Code Changes
- Changed the Index.cshtml for ProductType where the ProductType name is a link to the details page
- Changed the Details.cshtml for ProductType to list the products associated with that category and display the attributes of title, quantity and price
- Altered the ProductType controller Get Detail section to include Products

## Steps to Test

Outline the steps to test
```sh
git fetch --all
git checkout TG-2-ListProducts
```

examine areas of the ticket and ensure that everything works properly
- Click Product Categories in the navigation bar
- Confirm that when you click on a Product Type Name (like Movies and TV) it redirects you to a list of Products in that category
- Confirm that the list of products displayed contains the Title, Quantity and Price

## Impacted Areas in Application

List general components of the application that this PR will affect:

* `Controllers/ProductTypesController.cs` Detail() Task had a .Include added
* `Views/ProductTypes/Index.cshtml` changed TypeName to a link to the Product Type Details Page
* `Views/ProductTypes/Details.cshtml` changed to show list of products

## Mentions @username

@figamus
@ricky-bruner
@markhale94
@taylorlgulley
@askingalot

## Definition of Done

1. The project must be fully documented. This includes the following:
	* Comments for classes and methods added
	* Comments for models, controllers, and views added
2. The testing steps must completely work per ticket requirements.
3. Andy must smile approvingly and smirk with satisfaction following the smile.
